src/tests/hotreload-integration-test.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from '../platform/chai-web.js';
+import {StubLoader} from '../runtime/testing/stub-loader';
+import {Manifest} from '../runtime/manifest';
+import {SlotComposer, SlotComposerOptions} from '../runtime/slot-composer';
+import {ModalityHandler} from '../runtime/modality-handler';
+import {MessageChannel} from '../runtime/message-channel.js';
+import {Arc} from '../runtime/arc.js';
+import {ArcId, Id, IdGenerator} from '../runtime/id.js';
+import {HeadlessSlotDomConsumer} from '../runtime/headless-slot-dom-consumer.js';
+import {ParticleExecutionContext} from '../runtime/particle-execution-context.js';
+import {Loader} from '../runtime/loader.js';
+import {RozSlotComposer} from '../runtime/testing/fake-slot-composer.js';
+
+class StubSlotComposer extends SlotComposer {
+  constructor(options: SlotComposerOptions = {}) {
+    if (options.modalityHandler === undefined) {
+      options.modalityHandler = ModalityHandler.createHeadlessHandler();
+    }
+    super({
+      rootContainer: {'root': 'root-context'},
+      ...options});
+  }
+}
+
+class StubWasmLoader extends Loader {
+  public reloaded = false;
+
+  async loadWasmBinary(spec): Promise<ArrayBuffer> {
+    if (this.reloaded) {
+      spec.implFile = 'build/tests/source/test-module-new.wasm';
+    } else {
+      spec.implFile = 'build/tests/source/test-module-old.wasm';
```

Comment:
Doesn't the implFile need to match the existing one when a reload occurs? If not, how does the PEC know to tear down the old particle?

Reply:
The container is supposed to be destroyed in the beginning of the reload process (and the `loadWasmBinary` function is called later on). However I just changed it such that the implFile won't be changed (the loader will just reload the new .wasm file without changing the implFile name), which I think might be better.